tests/e2e/specs/wcpay/shopper/shopper-checkout-multi-currency.spec.js

```diff
@@ -0,0 +1,147 @@
+ * Internal dependencies
+ */
+
+import { fillCardDetails, setupProductCheckout } from '../../../utils/payments';
+import { merchantWCP, shopperWCP } from '../../../utils';
+
+const ORDER_RECEIVED_ORDER_TOTAL_SELECTOR =
+	'.woocommerce-order-overview__total';
+const ORDER_HISTORY_ORDER_ROW_SELECTOR = '.woocommerce-orders-table__row';
+const ORDER_HISTORY_ORDER_ROW_NUMBER_COL_SELECTOR =
+	'.woocommerce-orders-table__cell-order-number';
+const ORDER_HISTORY_ORDER_ROW_TOTAL_COL_SELECTOR =
+	'.woocommerce-orders-table__cell-order-total';
+const ORDER_DETAILS_ORDER_TOTAL_SELECTOR =
+	'.woocommerce-table--order-details tfoot tr:last-child td';
+
+const placeOrderWithCurrency = async ( currency ) => {
+	try {
+		await setupProductCheckout(
+			config.get( 'addresses.customer.billing' ),
+			[ [ config.get( 'products.simple.name' ), 1 ] ],
+			currency
+		);
+		const card = config.get( 'cards.basic' );
+		await fillCardDetails( page, card );
+		await shopper.placeOrder();
+		await expect( page ).toMatch( 'Order received' );
+
+		const url = await page.url();
+		return url.match( /\/order-received\/(\d+)\// )[ 1 ];
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error(
+			`Error placing order with currency ${ currency }: `,
+			error
+		);
+		throw error;
+	}
+};
+
+const getOrderTotalTextForOrder = async ( orderId ) => {
+	return await page.$$eval(
+		ORDER_HISTORY_ORDER_ROW_SELECTOR,
+		( rows, currentOrderId, orderNumberColSelector, totalColSelector ) => {
+			const orderSelector = `${ orderNumberColSelector } a[href*="view-order/${ currentOrderId }/"]`;
+			return rows
+				.filter( ( row ) => row.querySelector( orderSelector ) )
+				.map( ( row ) =>
+					row.querySelector( totalColSelector )?.textContent.trim()
+				)
+				.find( ( text ) => text !== null );
+		},
+		orderId,
+		ORDER_HISTORY_ORDER_ROW_NUMBER_COL_SELECTOR,
+		ORDER_HISTORY_ORDER_ROW_TOTAL_COL_SELECTOR
+	);
+};
+
+describe( 'Shopper Multi-Currency checkout', () => {
+	const currenciesOrders = {
+		USD: null,
+		EUR: null,
+	};
+	beforeAll( async () => {
+		// Enable multi-currency
+		await merchant.login();
+
+		await merchantWCP.activateMulticurrency();
+		for ( const currency in currenciesOrders ) {
+			await merchantWCP.addCurrency( currency );
+		}
+
+		await merchant.logout();
+
+		await shopper.login();
+	} );
+
+	afterAll( async () => {
+		await shopperWCP.emptyCart();
+		await shopper.logout();
+
+		// Disable multi-currency
+		await merchant.login();
+		await merchantWCP.deactivateMulticurrency();
+		await merchant.logout();
+	} );
+
+	describe.each( Object.keys( currenciesOrders ) )(
+		'Checkout process with %s currency',
+		( testCurrency ) => {
+			it( 'should allow checkout', async () => {
+				currenciesOrders[ testCurrency ] = await placeOrderWithCurrency(
+					testCurrency
+				);
+			} );
+
+			it( 'should display the correct currency on the order received page', async () => {
+				// should show correct currency in order received page
```

Comment:
Nit: Should we remove this comment since it restates the E2E test description?

Reply:
That was a leftover from the development phase 😄. It's been cleaned up now. https://github.com/Automattic/woocommerce-payments/pull/7804/commits/8f55857eb69163a0dda04ec7c7c4688a53369dfc